routers/api/v1/repo/issue.go

```diff
@@ -22,6 +23,123 @@ import (
 	milestone_service "code.gitea.io/gitea/services/milestone"
 )
 
+// SearchIssues searches for issues across the repositories that the user has access to
+func SearchIssues(ctx *context.APIContext) {
+	// swagger:operation GET /repos/issues/search issue issueSearchIssues
+	// ---
+	// summary: Search for issues across the repositories that the user has access to
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: state
+	//   in: query
+	//   description: whether issue is open or closed
+	//   type: string
+	// - name: labels
+	//   in: query
+	//   description: comma separated list of labels. Fetch only issues that have any of this labels. Non existent labels are discarded
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page number of requested issues
+	//   type: integer
+	// - name: q
+	//   in: query
+	//   description: search string
+	//   type: string
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/IssueList"
+
+	// find repos user can access
+	repoIDs := make([]int64, 0)
+	issueCount := 0
+	for page := 1; ; page++ {
+		repos, count, err := models.SearchRepositoryByName(&models.SearchRepoOptions{
+			Page:        page,
+			PageSize:    models.RepositoryListDefaultPageSize,
+			Private:     true,
+			Keyword:     "",
+			OwnerID:     ctx.User.ID,
+			TopicOnly:   false,
+			Collaborate: util.OptionalBoolNone,
+			UserIsAdmin: ctx.IsUserSiteAdmin(),
+			UserID:      ctx.Data["SignedUserID"].(int64),
+			OrderBy:     models.SearchOrderByID,
```

Comment:
I'm not sure I understand...do you mean we shouldn't search for issues across all the repositories the user has access to, but rather search across the let's say 15 most recently updated repositories that the user has access to? I could imagine some users thinking this is a bug, like I want to add a dependency to an issue in repository X, but the search won't list issues from the repository X (because the user has access to more than 15 repositories and repository X hasn't been updated in a while). Sorry if I'm just missing the point, thanks.

Reply:
We need to limit somehow the burden on the database for actions like this. Think of a bigger gitea installation, with hundreds of public repositories. From the user's point of view, it's also useless to have 100's of results and the one they want is on page 34. If we have to limit them in some way (and we must), chances are that dependencies are from the most active repositories.

There are other ways to address this problem, but they will demand more work. I'm just trying to find the shortest path to get this PR added to the main code. As I've said, there's always the possibility of a future PR with more enhancements. For example, we could let the user type in the issue in a text/list combo box, instead of using a list box. Other possibility is to have 15 as a default and let the admin change the value in `app.ini`.

Reply:
As a user, if I'm searching for an issue that I know is out there, I want it to show up, regardless of age.  I'm in favor of prioritizing the list by recently active but I think only searching on the recent issues would be a mistake.

If I search for the exact title of an issue in the 16th most active repository, what happens?

Also, how do we *know* that searching 10,000 issues will cause performance issues?

I could certainly live with an `app.ini` setting as we could increase it to a high number.  However, just noting that I think the whole premise will yield unexpected results for a user.

Reply:
thanks @guillep2k for your thoughts. I like the idea of using an app.ini setting to default the setting but allows users who want a deeper search to increase the value.

Reply:
@bhalbright if you'd like to go down that road, I'd suggest a setting that is the max number of repositories, 0 for none (it doesn't cost much to do it) and -1 for all of them. In any way, I'd use a lowish value as default (10~15).

Reply:
Hi, @guillep2k I wanted to run a different idea by you to see if you like it. Rather than implementing a setting in app.ini for a "repository search depth", how about instead implementing a setting in app.ini to enable "cross-repository dependencies". If enabled, when you search for issues to add as a dependency, it will search across all the repositories the user is granted access to (the way I have it implemented now). But if turned off, the search will work as it is today (search only the current repository). IMO this will be a little easier to implement. If a larger gitea installation isn't happy with the performance, they can just turn the feature off.

Reply:
I went ahead for now and implemented like I mentioned (either having cross-repo dependencies on or off), hope you don't mind. It seems to be working, but I'm going to test some more before I push up the commits

Reply:
I've commited these changes now (using a setting to turn on/off cross repo dependencies)